### ISSUE TITLE
doxygenfunction: Cannot find function (function(s) in a struct)

### ISSUE THREAD

**Issue by guyguy333** [michaeljones/breathe#219]
Hi,

Generating documentation using Breathe for the following constructors in a struct:

```cpp
template <typename T>
struct MyClass
{
  // Empty constructor
  MyClass();

  // Copy constructor
  MyClass(const MyClass&);

...
}
```

using:

```rst
.. doxygenfunction:: MyClass::MyClass
```
or
```rst
.. doxygenfunction:: MyClass< T >::MyClass
```
or
```rst
.. doxygenfunction:: MyClass<T>::MyClass
```
always result with the following warning:

```bash
WARNING: doxygenfunction: Cannot find function
```

The result is the same with the copy constructor (adding (const MyClass&))

Any ideas ?

Thank you

**Comment by michaeljones** [comment]
Hi,

Thanks for reporting this. I'm not entirely sure what is going wrong in Breathe there. I agree that ideally one of those two syntaxes would work. If you have time, I would really appreciate you providing a pull request with this as a failing example. Adding the code to the `examples/specific` folder and referencing from one of the test pages so show it failing. I will try to look into it though!

Cheers,
Michael

**Comment by guyguy333** [comment]
Sure, I opened a pull request with a failing example.

Thank you,

Guillaume

**Comment by michaeljones** [comment]
There is a fix for this on the `struct-function` branch. Happy to merge it into `master` if you're happy with the results.

**Comment by vitaut** [comment]
@michaeljones Do you plan to merge the `struct-function` branch?

**Comment by michaeljones** [comment]
I would be keen to. I encountered some weird behaviour which lead to an extra commit on the `whitespace` branch. I hope it isn't necessary though I can't explain what I was seeing. Could you give `struct-function` a go and we'll rebase & merge if it behaves for you?

**Comment by vitaut** [comment]
I've just tested `struct-function` branch on

```c++
/** Test */
template <typename T>
struct MyClass
{
  /** Test */
  MyClass();

  // Copy constructor
  MyClass(const MyClass&);
}
```

with

```rst
.. doxygenfunction:: MyClass::MyClass
```

and it worked as expected rendering `MyClass::MyClass()`, so it would be nice to see it merged.

**Comment by vitaut** [comment]
Fixed in #256.
